src/openrct2/actions/RideEntranceExitPlaceAction.hpp

```diff
@@ -0,0 +1,338 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../actions/RideEntranceExitRemoveAction.hpp"
+#include "../management/Finance.h"
+#include "../ride/Ride.h"
+#include "../ride/Station.h"
+#include "../world/Entrance.h"
+#include "../world/MapAnimation.h"
+#include "../world/Sprite.h"
+#include "GameAction.h"
+
+DEFINE_GAME_ACTION(RideEntranceExitPlaceAction, GAME_COMMAND_PLACE_RIDE_ENTRANCE_OR_EXIT, GameActionResult)
+{
+private:
+    CoordsXY _loc;
+    Direction _direction;
+    NetworkRideId_t _rideIndex;
+    uint8_t _stationNum;
+    bool _isExit;
+
+public:
+    RideEntranceExitPlaceAction() = default;
+
+    RideEntranceExitPlaceAction(CoordsXY loc, Direction direction, ride_id_t rideIndex, uint8_t stationNum, bool isExit)
+        : _loc(loc)
+        , _direction(direction)
+        , _rideIndex(rideIndex)
+        , _stationNum(stationNum)
+        , _isExit(isExit)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags();
```

Comment:
I couldn't find any place where the GAME_COMMAND_ALLOW_WHILE_PAUSED flag was set conditionally, could add it here and ditch the passing of the flag.

Reply:
You shouldn't be able to place a entranceexit when in paused but you should be able to show a ghost. I think thats the reason behind this.

Reply:
Now that you mention that, I wonder if we should always add the allow paused action flag if the ghost flag is specified.

Reply:
Would have to run some tests but I think that makes sense. Maybe we need to have some sort of central location that specifies all the implications each flag has.

Reply:
Agreed